build-tools/packages/build-tools/src/bumpVersion/bumpVersion.ts

```diff
@@ -123,29 +123,122 @@ function translateVirtualVersion(
+        }
+        await pkg.savePackageJson();
+    }
+
+    // for (const pkg of packageNeedBump) {
+    //     console.log(`  Bumping ${pkg.name}${vPatchLogString}`);
+    //     // Translate the versionBump into the appropriate change for virtual patch versioning
+    //     const translatedVersionBump = translateVirtualVersion(versionBump, versionBag.get(pkg.name), virtualPatch);
+    //     let cmd = `npm version ${translatedVersionBump}`;
+    //     if (pkg.getScript("build:genver")) {
+    //         cmd += " && npm run build:genver";
+    //     }
+    //     await exec(cmd, pkg.directory, `bump version on ${pkg.name}`);
+    // }
+
+    // Package json has changed. Reload.
+    return context.collectVersions(true);
+}
+
 /**
  * Bump version of packages in the repo
  *
  * @param versionBump the kind of version bump
  */
 export async function bumpRepo(
     context: Context,
-    versionBump: VersionChangeType,
+    versionBump: VersionChangeTypeExtended,
     monoReposNeedBump: Set<MonoRepoKind>,
     packageNeedBump: Set<Package>,
     virtualPatch: boolean,
-    versionBag: VersionBag,
+    versionBag?: VersionBag
 ) {
-    const bumpMonoRepo = async (repoVersionBump: VersionChangeType, monoRepo: MonoRepo) => {
+    const getVersion = (key: MonoRepoKind | string): string => {
+        let ver = "";
+        if (versionBag !== undefined && !versionBag.isEmpty()) {
+            for (const k of Object.keys(versionBag)) {
+                console.log(k);
+            }
+            ver = versionBag.get(key);
+        } else if (isMonoRepoKind(key)) {
+            // console.log(`getting version from repo`)
+            const repo = context.repo.monoRepos.get(key);
+            if (repo === undefined) {
+                fatal(`repo not found: ${key}`);
+            }
+            ver = repo.version;
+        } else {
+            fatal(`${key} is not a valid MonoRepoKind`);
+        }
+        return ver;
+    }
+
+    const bumpMonoRepo = async (repoVersionBump: VersionChangeType | semver.SemVer, monoRepo: MonoRepo) => {
         return exec(`npx lerna version ${repoVersionBump} --no-push --no-git-tag-version -y && npm run build:genver`, monoRepo.repoPath, "bump mono repo");
     }
 
     const vPatchLogString = virtualPatch ? " using virtual patches" : "";
 
     for (const monoRepo of monoReposNeedBump) {
-        console.log(`  Bumping ${monoRepo} version${vPatchLogString}`);
+        console.log(`  Bumping ${monoRepo}${vPatchLogString}...`);
         // Translate the versionBump into the appropriate change for virtual patch versioning
-        const translatedVersionBump = translateVirtualVersion(versionBump, versionBag.get(monoRepo), virtualPatch);
+        const ver = getVersion(monoRepo);
+        assert(ver, "ver is missing");
+        const adjVer = await adjustVersion(ver, versionBump);
+        const translatedVersionBump = adjVer!;
+        // translateVirtualVersion(adjVer!, versionBag.get(monoRepo), virtualPatch);
```

Comment:
We still use the virtual patch format in some places, so probably need to keep this call or move it to adjustVersion.